### PR TITLE
Add support for dispatch.then

### LINKED ISSUE #14: Can't call `then` on a debounced thunk dispatch
I'm having a problem chaining the results of dispatch.

According to redux-thunk docs, one should be able to call then on on the dispatch function,
e.g.,

```js
dispatch({action, type}).then(() => {/*it resolved!*/})
```

With this middleware in the mix, the dispatch function is returning undefined instead of a promise, so the call fails. If I pull out the meta value from the action, I'm able to call then on the dispatch as expected.

I tried modifying the lib/index.js directly in my node_modules directory and I found that it works fine if a promise is returned, i.e.:

```js
if (!cancel) {
  return new Promise((resolve) => {
    timers[key] = setTimeout(function () {
      resolve(dispatch(action));
    }, time);
  })
}

```
I'm not sure if there are any other considerations to explicitly returning a promise in cases where redux-thunk is not in use, but this seems to work for my use case.

### PR DESCRIPTION
**Pull request by tswaters**
Fixes #14

### CODE REVIEW

**Comment by ryanseddon** [comment]
Thanks could you add a test for this to expect a promise object to be returned

**Comment by tswaters** [comment]
Hi sorry initial commit I missed the new file, push forced so you should see it in `test/thunk.js`

**Comment by tswaters** [comment]
Actually looking at it now looks like I messed something up, the assertion is wrong.  I don't think it is actually performing that assertion.  On my way home now on mobile, will fix this later tonight probably

**Comment by tswaters** [comment]
@ryanseddon ok, should be ready now.  

In the initial PR, I forgot to actually call `resolve`, ha!  

This one test is structured slightly differently due to needing to call `then` on the dispatch.  It uses the `done` parameter of `it` to ensure the assertion is called.  Also, I've also moved the test to the same file to take advantage of the same `beforeEach` block.  

Let me know if this needs anything else anywhere, thanks

File: test/index.js [hunk 1]
```diff
@@ -186,4 +190,27 @@ describe('debounce middleware', () => {
       assert.deepEqual(store.getState(), {});
     });
   });
+
+
+  describe('using redux-thunk', () => {
+
```
Comment (ryanseddon) [inline test/index.js]: Nit pick drop this new line

File: test/index.js [hunk 2]
```diff
@@ -186,4 +190,27 @@ describe('debounce middleware', () => {
...
+      }
+    };
+
+    it('should work calling `then` on dispatch', done => {
+      store.dispatch(action)
+        .then(() => {
+          assert.deepEqual(store.getState(), {increment: 1});
+          done()
+        })
+        .catch(err => done(err));
```
Comment (ryanseddon) [inline test/index.js]: new line below this

File: test/index.js [hunk 3]
```diff
@@ -186,4 +190,27 @@ describe('debounce middleware', () => {
...
+      meta: {
+        debounce: {time:300}
+      }
+    };
+
+    it('should work calling `then` on dispatch', done => {
+      store.dispatch(action)
+        .then(() => {
+          assert.deepEqual(store.getState(), {increment: 1});
+          done()
```
Comment (ryanseddon) [inline test/index.js]: semicolon

File: test/index.js [hunk 4]
```diff
@@ -186,4 +190,27 @@ describe('debounce middleware', () => {
...
+        })
+        .catch(err => done(err));
+      clock.tick(300);
+    });
+  });
+
+
```
Comment (ryanseddon) [inline test/index.js]: Kill these new lines too

**Review by ryanseddon — APPROVED** [review]

**Comment by tswaters** [comment]
How's that?

edit:  err, missed one.... hold on.

File: test/index.js [hunk 5]
```diff
@@ -186,4 +190,25 @@ describe('debounce middleware', () => {
...
+        .then(() => {
+          assert.deepEqual(store.getState(), {increment: 1});
+          done();
+        })
+        .catch(err => done(err));
+
+      clock.tick(300);
+    });
+  });
+
```
Comment (ryanseddon) [inline test/index.js]: And this new line too

Comment (tswaters) [reply test/index.js]: done.

**Comment by tswaters** [comment]
Wicked, thanks man

**Comment by ryanseddon** [comment]
Should be on npm now as v0.4.0

**Pull request merged**